src/inmanta/references.py

```diff
@@ -122,6 +123,20 @@ def get_arg_value(
         return self.value
 
 
+class JsonArgument(Argument):
```

Comment:
Why make this a separate argument type? It's just another literal, isn't it? Not opposed, but I don't understand the motivation.

Reply:
I was a bit in doubt myself. I like the simplicity of literal (it is very definite in what is can contain). 

But it is not really a requirement (it's not like we will ever want to build a simpler decoder with support for only some types) . This is probably marginally faster in decoding invalid inputs, which also is not an argument. 

So I have no strong preference, if you want, I change it.

Reply:
No strong preference either, I just wondered.